nifi-nar-bundles/nifi-extension-utils/nifi-reporting-utils/src/main/java/org/apache/nifi/reporting/util/provenance/ProvenanceEventConsumer.java

```diff
@@ -256,9 +297,15 @@ private boolean isFilteringEnabled() {
                         }
                     }
                 }
+                if (!eventTypesExclude.isEmpty() && eventTypesExclude.contains(provenanceEventRecord.getEventType())) {
+                    continue;
```

Comment:
These two, `eventTypesExclude` and `eventTypes` are the most computationally cheap conditions. So, these should be done at the beginning, before checking ProcessGroup hierarchies. How do you think?

Reply:
Yes that's a good point, I just co-located them with their inclusionary counterparts. Will move them to the top.